vyper/semantics/types/function.py

```diff
@@ -577,3 +587,20 @@ def _generate_method_id(name: str, canonical_abi_types: List[str]) -> Dict[str,
     function_sig = f"{name}({','.join(canonical_abi_types)})"
     selector = keccak256(function_sig.encode())[:4].hex()
     return {function_sig: int(selector, 16)}
+
+
+def _annotate_nested_default_args(node, type_definition):
```

Comment:
looks a bit out of place. why aren't these annotated in `semantics/validation/annotation.py`?

Reply:
Currently, the node for the default value of an argument is annotated in the `from_FunctionDef` method of `ContractFunction`: https://github.com/vyperlang/vyper/blob/efe1dbef873af6995833d1220a3d56978dfb9de0/vyper/semantics/types/function.py#L337

I think this may tie in with adding `infer_arg_types` and `infer_kwarg_types` for all `Call` nodes?

Reply:
oh, i see, but can we use functionality from the annotation module instead of implementing something ad-hoc here?

Reply:
I think the closest would be to use `ExpressionAnnotationVisitor`, but that requires a `ContractFunction` to initialise, which is not available at this stage where we are trying to derive the `ContractFunction` from the `FunctionDef` node.

Reply:
I think the annotation module currently does not work for type definitions. In `vyper/semantics/validation/utils`, the `get_possible_types_from_node` function will throw an error if the node is a type definition.